lib/rules/best-practises/no-console.js

```diff
@@ -41,24 +41,31 @@ class NoConsoleLogChecker extends BaseChecker {
   ImportDirective(node) {
     this.isAnImport(node)
     if (this.isAnImport(node)) {
-      this.error(node, 'No import "hardhat/console.sol" or "forge-std/console.sol" statements') // eslint-disable-line
+      this.error(node, 'Unexpected import of console file')
     }
   }
 
   FunctionCall(node) {
     if (this.isConsoleLog(node)) {
-      this.error(node, 'No console.logX or console2.log statements')
+      this.error(node, 'Unexpected console statement')
     }
   }
 
   isConsoleLog(node) {
-    return node.expression.expression.name.includes('console')
+    return (
+      node.expression.expression.type === 'Identifier' &&
+      (node.expression.expression.name === 'console' ||
+        node.expression.expression.name === 'console2') &&
+      node.expression.memberName.startsWith('log')
+    )
   }
 
   isAnImport(node) {
     return (
       node.type === 'ImportDirective' &&
```

Comment:
I missed this in the previous review, but this check doesn't make sense anymore. Since you are using the `ImportDirective` callback, this check will always be true.

I think I would just inline this logic in the `ImportDirective` callback, this indirection is not necessary.

Reply:
Sorry I don't understand what you mean.

As I understood: 
ImportDirective is executed if/when the code has an import... so that is calling the isAnImport() function to check the import path

FunctionCall is executed if/when a call to a function is being done... like console.log, so that is why the expression makes sense

I did delete line 42 which was useless

Please post your suggestion or approve the pr if you consider we can move past this. Thanks!

Reply:
The `ImportDirective` callback is called when an import node is visited, so checking if it's of type 'ImportDirective' is redundant. I pushed a commit fixing it.

Reply:
ohhh sorry
I thought you were talking about these checks
```
node.expression.expression.name === 'console' ||
node.expression.expression.name === 'console2') &&
node.expression.memberName.startsWith('log')
```

that's why did not make sense for me...

Yes, it was useless to check if that was an importDirective since it is being called by that node